src/scripts/write-checksum.sh

```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+if [ ${PARAM_GRADLE_CACHE_CHECKSUM} != "/tmp/gradle_cache_seed" ]; then
+    echo '${PARAM_GRADLE_CACHE_CHECKSUM}' > ${PARAM_GRADLE_CACHE_CHECKSUM}.txt
```

Comment:
As you are using single quotes, the literal string will be saved in the text file.